packages/pass-style/src/byteArray.js

```diff
@@ -0,0 +1,55 @@
+import { X } from '@endo/errors';
+import { assertChecker } from './passStyle-helpers.js';
+
+const { getPrototypeOf, getOwnPropertyDescriptor } = Object;
+const { ownKeys, apply } = Reflect;
+
+// @ts-expect-error TODO How do I add it to the ArrayBuffer type?
+const AnImmutableArrayBuffer = new ArrayBuffer(0).transferToImmutable();
+/**
+ * As proposed, this will be the same as `ArrayBuffer.prototype`. As shimmed,
+ * this will be a hidden intrinsic that inherits from `ArrayBuffer.prototype`.
+ * Either way, get this in a way that we can trust it after lockdown, and
+ * require that all immutable ArrayBuffers directly inherit from it.
+ */
+const ImmutableArrayBufferPrototype = getPrototypeOf(AnImmutableArrayBuffer);
+
+// @ts-expect-error ok to implicitly assert the access is found
+const immutableGetter = getOwnPropertyDescriptor(
+  ImmutableArrayBufferPrototype,
+  'immutable',
+).get;
+
+/**
+ * @param {unknown} candidate
+ * @param {import('./types.js').Checker} [check]
+ * @returns {boolean}
+ */
+const canBeValid = (candidate, check = undefined) =>
+  (candidate instanceof ArrayBuffer &&
+    // @ts-expect-error TODO How do I add it to the ArrayBuffer type?
+    candidate.immutable) ||
```

Comment:
Remind me, what are the semantics of `canBeValid`, and why are the checks much loser?

Reply:
Both better error messages and reducing search. First, what kind of passable does it seem the candidate is trying to be? This first checking pass should be quick. But the check ***must*** be tight enough to ensure that the candidate cannot be valid as any of the later categories. (Logic programming calls this committed choice.) Once we know what bucket it's in, we then try to see if it is a valid case of that bucket. If not, we can emit more relevant complaints, rather than continue searching.

Reply:
Resolving